scater-calculate-qc-metrics.R

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env Rscript 
+
+# Load optparse we need to check inputs
+suppressPackageStartupMessages(require(optparse))
+
+# Load common functions
+suppressPackageStartupMessages(require(workflowscriptscommon))
+
+# parse options
+option_list = list(
+  make_option(
+    c("-i", "--input-object-file"),
+    action = "store",
+    default = NA,
+    type = 'character',
+    help = "File name in which a serialized R SingleCellExperiment object where object matrix found"
+  ),
+  make_option(
+    c("-e", "--exprs-values"),
+    action = "store",
+    default = 'counts',
+    type = 'character',
+    help= "character string indicating which slot of the assayData from the 'SingleCellExperiment' object should be used to compute log-transformed expression values. Valid options are 'counts', 'tpm', 'cpm' and 'fpkm'. Defaults to the first available value of the options in the order shown."
+  ),
+  make_option(
+    c("-f", "--feature-controls"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "comma-separated list of files with one feature per line. A named list is created containing one or more vectors to identify feature controls (for example, ERCC spike-in genes, mitochondrial genes, etc)"
+  ),
+  make_option(
+    c("-c", "--cell-controls"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "a character vector of cell (sample) names, or a logical vector, or a numeric vector of indices used to identify cell controls (for example, blank wells or bulk controls)."
```

Comment:
This probably also needs to be provided via a file as per breakdown document (though I didn't reflect that in the description text - my bad)